experiments/integrated/fluxnet/data_tools.jl

```diff
@@ -93,35 +72,35 @@ function VerifiedColumn(file_matrix::Matrix, column_name::String, units::String)
     col_dat, status = check_column(file_matrix, column_name)
     if status == absent
         return DataColumn(reshape([], 0, 0), "", status)
-    end
-    # Check for a QC flag column
-    column_QC = column_name * "_QC"
-    QC_col, QCstatus = check_column(file_matrix, column_QC)
-    # If no QC flag, then fill and return the data column as is
-    if QCstatus != absent && minimum(QC_col) != 2
-        # Fill any missing data with the mean value per the QC flag
-        num_missing = count(x -> x > 1, QC_col)
-        if num_missing > 0
-            perc_missing = 100.0 * num_missing / length(col_dat)
-            @info "Warning: Data for $column_name $(
-            round(perc_missing, sigdigits=3))% missing. Filling with mean value"
-            replace_missing_with_mean!(col_dat, QC_col)
+    elseif status == complete
+        return DataColumn(col_dat, units, status)
```

Comment:
One thing here is that we need to check for the QC flag and fill using the QC flag before we can consider the column complete. `check_column` only checks for missings by value (`==-9999`), but in the datasets there are values that are missing or very low quality according to the QC flag but actually have data populated in the normal range. As a result, `check_column` can return status complete when the QC flag will say something different.

Alternatively we could add checking QC to the `check_column` function to neaten the logic here

Reply:
My original goal here was that `check_column` just ensured that column existed in the file, and then `filter_column` processed it for missing values which is why I didn't include the QC in `check_column`

Reply:
ah ok! Ill try and adjust this to reflect that, and then re-request review